Source/Core/Core/TimePlayed.cpp

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+#include <string>
+
+#include "Common/FileUtil.h"
+#include "Common/IniFile.h"
+#include "TimePlayed.h"
+
+// used for QT interface - general access to time played for games
+TimePlayed::TimePlayed()
+{
+  m_game_id = "None";
+  ini_path = File::GetUserPath(D_CONFIG_IDX) + "TimePlayed.ini";
+  ini.Load(ini_path);
+  time_list = ini.GetOrCreateSection("Time Played");
+}
+
+void FilterUnsafeCharacters(std::string& game_id)
+{
+  const std::string forbiddenChars = "\\/:?\"<>|";
+  for (auto& chr : game_id)
+  {
+    if (forbiddenChars.find(chr) != std::string::npos)
+    {
+      chr = '_';
+    }
+  }
+}
```

Comment:
Can you maybe use `Common::EscapeFileName()` (in NandPaths.h) for this? That way multiple IDs won't match onto the same 'filtered' ID, and the original ID is recoverable if needed.